File: backend/app/schemas/shopping_list.py
from __future__ import annotations

from datetime import date, datetime

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.daily_menu import DailyMenuPublic


class ShoppingListItemPublic(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    amount: str
    unit: str
    is_purchased: bool


class ShoppingListPublic(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    family_id: int
    menu_date: date
    created_at: datetime
    items: list[ShoppingListItemPublic]


class ShoppingListTodayResponse(BaseModel):
    shopping_list: ShoppingListPublic | None = None
    menu: DailyMenuPublic | None = None


class ShoppingListItemUpdateRequest(BaseModel):
    is_purchased: bool


class ShoppingListItemResponse(BaseModel):
    item: ShoppingListItemPublic


class ShoppingListResetResponse(BaseModel):
    shopping_list: ShoppingListPublic | None = None

File: frontend/types/shopping-list.ts
import type { DailyMenu } from "@/types/daily-menu";

export type ShoppingListItem = {
  id: number;
  name: string;
  amount: string;
  unit: string;
  is_purchased: boolean;
};

export type ShoppingList = {
  id: number;
  family_id: number;
  menu_date: string;
  created_at: string;
  items: ShoppingListItem[];
};

export type ShoppingListTodayResponse = {
  shopping_list: ShoppingList | null;
  menu: DailyMenu | null;
};

export type ShoppingListItemUpdatePayload = {
  is_purchased: boolean;
};

export type ShoppingListItemResponse = {
  item: ShoppingListItem;
};

export type ShoppingListResetResponse = {
  shopping_list: ShoppingList | null;
};
